app/services/GuidanceService.scala

```diff
@@ -183,18 +187,25 @@ class GuidanceService @Inject() (
         Future.successful(Left(err))
       case Right(process) =>
         logger.warn(s"Loaded process ${process.meta.id}, containing ${process.flow.keys.toList.length} stanzas, ${process.phrases.length} phrases")
-        pageBuilder.pages(process, process.startPageId).fold(
-        err => {
+        pageBuilder.pages(process, process.startPageId).fold(err => {
           logger.warn(s"Failed to parse process with error $err")
           Future.successful(Left(InvalidProcessError))
         },
-        pages => sessionRepository.set(docId, process, pages.map(p => p.url -> p.id).toMap).map {
-          case Right(_) => Right((pages.head.url, process.meta.processCode))
-          case Left(err) =>
-            logger.error(s"Failed to store new parsed process in session repository, $err")
-            Left(err)
+        pages => {
+          val urlMap: Map[String, String] = pages.map(p => (p.id, p.url)).toMap
```

Comment:
As you are doing a bit of work here it would be a good idea to put the debug logging inside an if block as follows

if(logger.isDebugEnabled()) {
.....
.....
}
Then when you aren't debugging the code won't get executed

Reply:
done